### PR TITLE
Nit fixes on documentation

### PR DESCRIPTION
**Pull request by rajarshimaitra**
### Description

This builds on #793  and adds documentation fixes for the alpha release. It mainly includes typos, sentence constructions, and punctuations. Fixes are reported by Grammarly. Only fixed files are modules added from bdk_core. Previous bdk modules are untouched. 

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by notmandatory** [comment]
This can be rebased now to `master` since #793 is merged.

File: nursery/coin_select/src/coin_selector.rs [hunk 1]
```diff
@@ -529,7 +528,7 @@ mod test {
 
     use super::{CoinSelector, CoinSelectorOpt, WeightedValue};
 
-    /// Ensure `target_value` is respected. Can't have no disrespect.
+    /// Ensure `target_value` is respected. Can't have any disrespect.
```
Comment (evanlinjin) [inline nursery/coin_select/src/coin_selector.rs:531]: Oh come on.

**Review by evanlinjin — APPROVED** [review]
ACK 24df03afd6f4f5584cf1421b36209ac7a9f93608

Thanks for the work! I rebased this on top of master also.

**Review by rajarshimaitra — COMMENTED** [review]
I think @evanlinjin already rebased.. 

ACK 24df03afd6f4f5584cf1421b36209ac7a9f93608

**Pull request merged**